internal/controller/clusterinstance_controller.go

```diff
@@ -325,12 +347,12 @@ func (r *ClusterInstanceReconciler) renderManifests(
 		}
 	}
 
-	return renderedManifests, err
+	return renderedManifestObjs, err
 }
 
 // getSyncWave extracts the syncWave from the given object manifest
 // if the syncWave cannot be parsed, a nil-int pointer is returned with the error
-func getSyncWave(object interface{}) (*int, error) {
+func getSyncWave(iObject interface{}) (*int, error) {
```

Comment:
What is an `iObject`? Why change this name?

Reply:
Alternatively, why can't this function parameter be a `RenderedObject` type directly?

Reply:
Also why does this return an int pointer? If there's an error returning `0` would be fine.

Reply:
> Alternatively, why can't this function parameter be a RenderedObject type directly?
Great suggestion! Will change this :)

Reply:
I had already defined a GetSyncWave function to the RenderedObject type 🤦‍♂️